src/main/java/com/synopsys/integration/alert/common/field/JsonField.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (C) 2018 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.field;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Optional;
+
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.TypeRef;
+import com.synopsys.integration.alert.common.enumeration.FieldContentIdentifier;
+import com.synopsys.integration.util.Stringable;
+
+public class JsonField<T> extends Stringable {
+    public static final String LABEL_URL_SUFFIX = "_url";
+
+    private final TypeRef<?> typeRef;
+    private final FieldContentIdentifier contentIdentifier;
+    private final JsonPath configNameMapping;
+    private final JsonPath jsonPath;
+
+    private final String fieldName;
+    private final String label;
+
+    public static <T> JsonField<T> createObjectField(final JsonPath jsonPath, final String fieldName, final FieldContentIdentifier contentIdentifier, final String label, final TypeRef<?> typeRef) {
+        return new JsonField<T>(typeRef, jsonPath, fieldName, contentIdentifier, label);
+    }
+
```

Comment:
Because we set the configuration of JsonPath to always return lists, the TypeRef must always be a List of the type you want